Description: Add support for "twolame" mp3 encoder.
Author: Cristian Greco <cristian.debian@gmail.com>

--- a/pacpl
+++ b/pacpl
@@ -537,6 +537,17 @@
                                         PROMPT => { BITRATE => 1 },
                                       },
 
+                           twolame => {
+                                        NAME => "twolame",
+                                        ESTR => sub {
+                                                      $opts = "-b $bitrate" if $defopts == 1;
+                                                      $opts = '' if $defopts == 0;
+                                                      "$eopts $opts %i %o"
+                                                    },
+
+                                        PROMPT => { BITRATE => 1 },
+                                      },
+
                             ffmpeg => {
                                         NAME => "ffmpeg",
                                         ESTR => sub { 
