Index: pacpl-4.0.5/codecs.conf
===================================================================
--- pacpl-4.0.5.orig/codecs.conf	2010-03-20 20:37:37.000000000 +0100
+++ pacpl-4.0.5/codecs.conf	2010-03-20 20:38:14.000000000 +0100
@@ -1,5 +1,5 @@
 # specify the default encoder/decoder for each file format.
-# 
+#
 # FORMAT = encoder,decoder <-- must look exactly like this.
 #
 #
@@ -18,6 +18,10 @@
 #
 AAC = faac,faad
 
+# encoders: jmac, mac
+# decoders: jmac, mac
+APE = jmac,jmac
+
 # encoders: flac, ffmpeg
 # decoders: flac, ffmpeg, mplayer
 #
Index: pacpl-4.0.5/pacpl
===================================================================
--- pacpl-4.0.5.orig/pacpl	2010-03-20 20:37:44.000000000 +0100
+++ pacpl-4.0.5/pacpl	2010-03-20 20:38:51.000000000 +0100
@@ -38,7 +38,7 @@
 use CDDB_get qw(get_cddb);
 
 # non-encoder/decoder related options
-my 
+my
 (
    $to,
    $recursive,
@@ -140,7 +140,7 @@
 load_lang();
 
 # try to load configuration file in this order...
-my @conf_locations = ( 
+my @conf_locations = (
                        "$ENV{HOME}/.pacplrc",   # Local
                        "$conf_path/pacpl.conf", # Global
                        "$ENV{PWD}/pacpl.conf",  # Current Directory
@@ -181,20 +181,20 @@
 sub load_codecs {
 
     open(CODECS, "$codecs_conf") or die "$lang{opening_file} $codecs_conf - $!\n";
-        
+
     my @codecs = <CODECS>;
        @codecs = grep(!/^$|^#/, @codecs);
-       
+
     close(CODECS);
-        
+
        foreach (@codecs) {
-        
+
             chomp;
             my ($k, $v) = split(/\s+=\s+/);
             my ($e, $d) = split(/,/, $v);
-            
+
                 $k =~ tr/A-Z/a-z/;
-            
+
                 $run{$k}{DEFAULT_ENCODER} = $e if (exists($run{$k}{DEFAULT_ENCODER}));
                 $run{$k}{DEFAULT_DECODER} = $d if (exists($run{$k}{DEFAULT_DECODER}));
        }
@@ -204,27 +204,27 @@
 sub load_lang {
 
     my $po = "$po_dir/$ENV{LANG}.po";
-       $po =~ s/\.UTF-8//i      if $po =~ /UTF-8/i; 
-       $po =~ s/\.utf8//i       if $po =~ /utf8/i;  
-       $po =~ s/_\w+//          if not -e $po;      
-       $po = "$po_dir/en_US.po" if not -e $po;      
-       
+       $po =~ s/\.UTF-8//i      if $po =~ /UTF-8/i;
+       $po =~ s/\.utf8//i       if $po =~ /utf8/i;
+       $po =~ s/_\w+//          if not -e $po;
+       $po = "$po_dir/en_US.po" if not -e $po;
+
     open(LANG, "< $po") or die "error: could not open - $po\n$!\n";
-    
+
     my @data = <LANG>;
        @data = grep(!/^#|^$/, @data);
-       
+
     close(LANG);
-    
+
     foreach (@data) {
-         chomp;    
+         chomp;
          s/=/__EQ__/;
          my ($k, $v) = split(/\s+__EQ__\s+/, $_);
          $lang{$k} = $v;
     }
-    
+
 }
-    
+
 my $banner = 1;
 
 # print error messages
@@ -245,7 +245,7 @@
     print "$lang{$key} $msg";
     print "\n"    if $nm == 1;
     print "\n\n"  if $nm == 2;
-      
+
     system("kdialog --title \"$name\" --passivepopup \"$lang{$key} $msg\" 10 &") if $gui;
 
     $banner = 0;
@@ -271,7 +271,7 @@
 my $aacqual   = $config{AACQUAL};
 my $mpcqual   = $config{MPCQUAL};
 my $ofmode    = $config{OFMODE};
-my $ofopt     = $config{OFOPT};  
+my $ofopt     = $config{OFOPT};
 my $bratio    = $config{BRATIO};
 my $bquanl    = $config{BQUANL};
 my $bpsize    = $config{BPSIZE};
@@ -289,7 +289,7 @@
 
 # command line options
 GetOptions
-( 
+(
 	't|to=s'      => \$to,
 	'r|recursive' => \$recursive,
 	'p|preserve'  => \$preserve,
@@ -309,7 +309,7 @@
 	'normalize'   => \$normalize,
 	'encoder=s'   => \$my_encoder,
 	'decoder=s'   => \$my_decoder,
-	
+
 	'title=s'     => \$title,
 	'track=n'     => \$track,
 	'artist=s'    => \$artist,
@@ -317,7 +317,7 @@
 	'comment=s'   => \$comment,
 	'year=n'      => \$year,
 	'genre=s'     => \$genre,
-	
+
 	'gui'         => \$gui,
 
 	'eopts=s'     => \$eopts,
@@ -343,7 +343,7 @@
 	'bratio=n'    => \$bratio,
 	'bquanl=n'    => \$bquanl,
 	'bpsize=n'    => \$bpsize,
-	
+
         'rip=s'       => \$rip,
 	'nocddb'      => \$nocddb,
 	'noinput'     => \$noinput,
@@ -355,7 +355,7 @@
 $silent = '' if $verbose;
 my $opts;
 
-# conversion options.  when adding a new format, 
+# conversion options.  when adding a new format,
 # wild card %i = input file and %o = output file
 %run = (
 
@@ -363,23 +363,23 @@
 
                 DEFAULT_ENCODER => "sox",
                 DEFAULT_DECODER => "sox",
-                
+
                 ENCODER => {
                               sox => {
                                       NAME => "sox",
-                                      ESTR => sub { 
+                                      ESTR => sub {
                                                     $opts = "-r $freq -c $channels" if $defopts == 1;
                                                     $opts = '' if $defopts == 0;
-                                                    "%i $opts $eopts %o $effect" 
+                                                    "%i $opts $eopts %o $effect"
                                                   },
-                                    
+
                                       PROMPT => {
                                                   FREQ     => 1,
                                                   CHANNELS => 1,
                                                 },
                                      },
                            },
-                           
+
                 DECODER => {
                               sox => {
                                         NAME => "sox",
@@ -398,7 +398,7 @@
 
                 DEFAULT_ENCODER => undef,
                 DEFAULT_DECODER => "ffmpeg",
-                
+
                 DECODER => {
                               ffmpeg => {
                                           NAME => "ffmpeg",
@@ -406,7 +406,7 @@
                                           PROMPT => { NONE => 0 },
                                         },
                            },
-                            
+
                 TAGS    => {
                              READ   => 0,
                              WRITE  => 0,
@@ -418,16 +418,16 @@
 
                 DEFAULT_ENCODER => "ffmpeg",
                 DEFAULT_DECODER => "mplayer",
-                
+
                 ENCODER => {
                               ffmpeg => {
                                           NAME => "ffmpeg",
-                                          ESTR => sub { 
+                                          ESTR => sub {
                                                         $opts = "-ab $bitrate -ar $freq -ac $channels" if $defopts == 1;
                                                         $opts = '' if $defopts == 0;
                                                         "$eopts -y -i %i $opts %o"
                                                       },
-                                          
+
                                           PROMPT => {
                                                       BITRATE  => 1,
                                                       FREQ     => 1,
@@ -435,7 +435,7 @@
                                                     },
                                         },
                            },
-                           
+
                 DECODER => {
                              mplayer => {
                                           NAME => "mplayer",
@@ -449,56 +449,56 @@
                              MODULE => undef,
                            },
                },
-               
+
          rm => {
-        
+
                  DEFAULT_ENCODER => 'ffmpeg',
                  DEFAULT_DECODER => 'mplayer',
-                 
+
                  ENCODER => {
                               ffmpeg => {
                                           NAME => 'ffmpeg',
-                                          ESTR => sub { 
+                                          ESTR => sub {
                                                         $opts = "-ar $freq -ac $channels" if $defopts == 1;
                                                         $opts = '' if $defopts == 0;
-                                                        "$eopts -y -i %i $opts %o" 
+                                                        "$eopts -y -i %i $opts %o"
                                                       },
-                                          
+
                                           PROMPT => {
                                                       FREQ     => 1,
                                                       CHANNELS => 1,
                                                     },
                                         },
                            },
-                           
+
                 DECODER => {
                              mplayer => {
                                           NAME => 'mplayer',
                                           DSTR => sub { "-vc dummy -vo null -nortc -ao pcm:file=%o %i" },
                                         },
                            },
- 
+
                    TAGS => {
                              READ   => 0,
                              WRITE  => 0,
                              MODULE => undef,
                            },
                },
-               
+
         mp3 => {
 
                 DEFAULT_ENCODER => "lame",
                 DEFAULT_DECODER => "lame",
-                
+
                 ENCODER => {
                               lame => {
                                         NAME => "lame",
-                                        ESTR => sub { 
+                                        ESTR => sub {
                                                       $opts = "--resample $freq -b $bitrate -h" if $defopts == 1;
                                                       $opts = '' if $defopts == 0;
-                                                      "$eopts $opts %i %o" 
+                                                      "$eopts $opts %i %o"
                                                     },
-                                      
+
                                         PROMPT => {
                                                     BITRATE => 1,
                                                     FREQ    => 1,
@@ -507,33 +507,33 @@
 
                           bladeenc => {
                                         NAME => "bladeenc",
-                                        ESTR => sub { 
+                                        ESTR => sub {
                                                       $opts = "-br $bitrate" if $defopts == 1;
                                                       $opts = '' if $defopts == 0;
-                                                      "$eopts $opts %i %o" 
-                                                    },                                                    
+                                                      "$eopts $opts %i %o"
+                                                    },
                                         PROMPT => { BITRATE => 1 },
                                       },
 
                               gogo => {
                                         NAME => "gogo",
-                                        ESTR => sub { 
+                                        ESTR => sub {
                                                       $opts = "-b $bitrate" if $defopts == 1;
                                                       $opts = '' if $defopts == 0;
-                                                      "$eopts $opts %i %o" 
+                                                      "$eopts $opts %i %o"
                                                     },
-                                        
+
                                         PROMPT => { BITRATE => 1 },
                                       },
 
                            toolame => {
                                         NAME => "toolame",
-                                        ESTR => sub { 
+                                        ESTR => sub {
                                                       $opts = "-b $bitrate" if $defopts == 1;
                                                       $opts = '' if $defopts == 0;
-                                                      "$eopts $opts %i %o" 
+                                                      "$eopts $opts %i %o"
                                                     },
-                                        
+
                                         PROMPT => { BITRATE => 1 },
                                       },
 
@@ -550,40 +550,40 @@
 
                             ffmpeg => {
                                         NAME => "ffmpeg",
-                                        ESTR => sub { 
+                                        ESTR => sub {
                                                       $opts = "-ab $bitrate.k -ar $freq -ac $channels" if $defopts == 1;
                                                       $opts = '' if $defopts == 0;
-                                                      "$eopts -y -i %i $opts %o" 
+                                                      "$eopts -y -i %i $opts %o"
                                                     },
-                                        
+
                                         PROMPT => {
                                                     BITRATE  => 1,
                                                     FREQ     => 1,
                                                     CHANNELS => 1,
                                                   },
                                       },
- 
+
                                sox => {
                                         NAME => "sox",
-                                        ESTR => sub { 
+                                        ESTR => sub {
                                                       $opts = "-r $freq -c $channels" if $defopts == 1;
                                                       $opts = '' if $defopts == 0;
-                                                      "%i $eopts $opts %o" 
+                                                      "%i $eopts $opts %o"
                                                     },
-                                        
+
                                         PROMPT => {
                                                     FREQ     => 1,
                                                     CHANNELS => 1,
                                                   },
                                       },
                            },
-                            
+
                 DECODER => {
-                              lame => { 
+                              lame => {
                                         NAME => "lame",
                                         DSTR => sub { "$dopts --decode %i %o" },
                                       },
-                             
+
                             ffmpeg => {
                                         NAME => "ffmpeg",
                                         DSTR => sub { "$dopts -y -i %i %o" },
@@ -611,16 +611,16 @@
 
                 DEFAULT_ENCODER => "oggenc",
                 DEFAULT_DECODER => "oggdec",
-                
+
                 ENCODER => {
                               oggenc => {
                                           NAME => "oggenc",
-                                          ESTR => sub { 
+                                          ESTR => sub {
                                                         $opts = "--resample $freq -q $oggqual" if $defopts == 1;
                                                         $opts = '' if $defopts == 0;
-                                                        "$eopts $opts %i -o %o" 
+                                                        "$eopts $opts %i -o %o"
                                                       },
-                                        
+
                                           PROMPT => {
                                                       FREQ    => 1,
                                                       OGGQUAL => 1,
@@ -629,12 +629,12 @@
 
                               ffmpeg => {
                                           NAME => "ffmpeg",
-                                          ESTR => sub { 
+                                          ESTR => sub {
                                                         $opts = "-ab $bitrate.k -ar $freq -ac $channels" if $defopts == 1;
                                                         $opts = '' if $defopts == 0;
-                                                        "$eopts -y -i %i $opts %o" 
+                                                        "$eopts -y -i %i $opts %o"
                                                       },
-                                          
+
                                           PROMPT => {
                                                       BITRATE  => 1,
                                                       FREQ     => 1,
@@ -644,19 +644,19 @@
 
                                  sox => {
                                           NAME => "sox",
-                                          ESTR => sub { 
+                                          ESTR => sub {
                                                         $opts = "-r $freq -c $channels" if $defopts == 1;
                                                         $opts = '' if $defopts == 0;
-                                                        "%i $eopts $opts %o" 
+                                                        "%i $eopts $opts %o"
                                                       },
-                                          
+
                                           PROMPT => {
                                                       FREQ     => 1,
                                                       CHANNELS => 1,
                                                     },
                                         },
                            },
-                           
+
                 DECODER => {
                               oggdec => {
                                           NAME => "oggdec",
@@ -685,31 +685,31 @@
                              MODULE => "Ogg::Vorbis::Header",
                            },
                },
-       
+
        spx  => {
-       
+
                 DEFAULT_ENCODER => "speexenc",
                 DEFAULT_DECODER => "speexdec",
-                
+
                 ENCODER => {
                               speexenc => {
                                             NAME => "speexenc",
-                                            ESTR => sub { 
+                                            ESTR => sub {
                                                           $opts = "--quality $spxqual" if $defopts == 1;
                                                           $opts = '' if $defopts == 0;
-                                                          "$eopts $opts %i %o" 
+                                                          "$eopts $opts %i %o"
                                                         },
                                             PROMPT => { SPXQUAL => 1 },
                                           },
                            },
-                           
+
                 DECODER => {
                               speexdec => {
                                             NAME => "speexdec",
                                             DSTR => sub { "$dopts %i %o" },
                                           },
                            },
-                
+
                 TAGS    => {
                              READ   => 1,
                              WRITE  => 1,
@@ -721,14 +721,14 @@
 
                 DEFAULT_ENCODER => "flac",
                 DEFAULT_DECODER => "flac",
-                
+
                 ENCODER => {
                               flac => {
                                         NAME => "flac",
-                                        ESTR => sub { 
+                                        ESTR => sub {
                                                       $opts = "-$fcomp" if $defopts == 1;
                                                       $opts = '' if $defopts == 0;
-                                                      "$eopts -f $opts %i -o %o" 
+                                                      "$eopts -f $opts %i -o %o"
                                                     },
                                         PROMPT => { FCOMP => 1 },
                                       },
@@ -742,15 +742,15 @@
                                                      },
                                         PROMPT => { FCOMP => 1 },
                                       },
-                                    
+
                             ffmpeg => {
                                         NAME => "ffmpeg",
-                                        ESTR => sub { 
+                                        ESTR => sub {
                                                       $opts = "-ab $bitrate.k -ar $freq -ac $channels" if $defopts == 1;
                                                       $opts = '' if $defopts == 0;
-                                                      "$eopts -y -i %i $opts %o" 
+                                                      "$eopts -y -i %i $opts %o"
                                                     },
-                                        
+
                                       PROMPT => {
                                                     BITRATE  => 1,
                                                     FREQ     => 1,
@@ -764,7 +764,7 @@
                                         NAME => "flac",
                                         DSTR => sub { "$dopts -f -d %i -o %o" },
                                       },
-                                    
+
                             ffmpeg => {
                                         NAME => "ffmpeg",
                                         DSTR => sub { "$dopts -y -i %i %o" },
@@ -787,19 +787,19 @@
 
                 DEFAULT_ENCODER => "flac",
                 DEFAULT_DECODER => "flac",
-                
+
                 ENCODER => {
                               flac => {
                                         NAME => "flac",
-                                        ESTR => sub { 
+                                        ESTR => sub {
                                                       $opts = "-$fcomp" if $defopts == 1;
                                                       $opts = '' if $defopts == 0;
-                                                      "$eopts -f $opts %i -o %o" 
+                                                      "$eopts -f $opts %i -o %o"
                                                     },
                                         PROMPT => { FCOMP => 1 },
                                       },
                            },
-                           
+
                 DECODER => {
                               flac => {
                                         NAME => "flac",
@@ -823,24 +823,34 @@
 
                 DEFAULT_ENCODER => "mac",
                 DEFAULT_DECODER => "mac",
-                
+
                 ENCODER => {
                               mac => {
                                         NAME => "mac",
-                                        ESTR => sub { 
+                                        ESTR => sub {
                                                       $opts = "-c$acomp" if $defopts == 1;
                                                       $opts = '' if $defopts == 0;
-                                                      "%i %o $opts $eopts" 
+                                                      "%i %o $opts $eopts"
                                                     },
                                         PROMPT => { ACOMP => 1 },
                                      },
+                              jmac => {
+                                        NAME => "java",
+                                        ESTR => sub {
+                                                      "-jar /usr/share/java/jmac.jar c2 %i %o"
+                                                    },
+                                     },
                             },
-                
+
                 DECODER => {
                               mac => {
                                         NAME => "mac",
                                         DSTR => sub { "%i %o -d $dopts" },
                                      },
+                              jmac => {
+                                        NAME => "java",
+                                        DSTR => sub { "-jar /usr/share/java/jmac.jar d %i %o" },
+                                     },
                            },
 
                 TAGS    => {
@@ -854,16 +864,16 @@
 
                 DEFAULT_ENCODER => "bonk",
                 DEFAULT_DECODER => "bonk",
-                
+
                 ENCODER => {
                               bonk => {
                                         NAME => "bonk",
-                                        ESTR => sub { 
+                                        ESTR => sub {
                                                       $opts = "-q $bquanl -b $bratio -s $bpsize" if $defopts == 1;
                                                       $opts = '' if $defopts == 0;
-                                                      "encode $eopts $opts %i -o %o" 
+                                                      "encode $eopts $opts %i -o %o"
                                                     },
-                                        
+
                                         PROMPT => {
                                                     BQUANL => 1,
                                                     BRATIO => 1,
@@ -871,7 +881,7 @@
                                                   },
                                       },
                            },
-                
+
                 DECODER => {
                               bonk => {
                                         NAME => "bonk",
@@ -890,19 +900,19 @@
 
                 DEFAULT_ENCODER => "shorten",
                 DEFAULT_DECODER => "shorten",
-                
+
                 ENCODER => {
                               shorten => {
                                             NAME => "shorten",
-                                            ESTR => sub { 
+                                            ESTR => sub {
                                                           $opts = "-c $channels" if $defopts == 1;
                                                           $opts = '' if $defopts == 0;
-                                                          "$eopts $opts %i %o" 
+                                                          "$eopts $opts %i %o"
                                                         },
                                             PROMPT => { CHANNELS => 1 },
                                          },
                            },
-                
+
                 DECODER => {
                               shorten => {
                                             NAME => "shorten",
@@ -921,26 +931,26 @@
 
                 DEFAULT_ENCODER => "faac",
                 DEFAULT_DECODER => "faad",
-                
+
                 ENCODER => {
                               faac => {
                                         NAME => "faac",
-                                        ESTR => sub { 
+                                        ESTR => sub {
                                                       $opts = "-q $aacqual" if $defopts == 1;
                                                       $opts = '' if $defopts == 0;
-                                                      "$eopts $opts %i -o %o" 
+                                                      "$eopts $opts %i -o %o"
                                                     },
                                         PROMPT => { AACQUAL => 1 },
                                       },
-                                      
+
                             ffmpeg => {
                                         NAME => "ffmpeg",
-                                        ESTR => sub { 
+                                        ESTR => sub {
                                                       $opts = "-ab $bitrate.k -ar $freq -ac $channels" if $defopts == 1;
                                                       $opts = '' if $defopts == 0;
-                                                      "$eopts -y -i %i $opts %o" 
+                                                      "$eopts -y -i %i $opts %o"
                                                     },
-                                        
+
                                         PROMPT => {
                                                     BITRATE  => 1,
                                                     FREQ     => 1,
@@ -948,13 +958,13 @@
                                                   },
                                       },
                            },
-                           
+
                 DECODER => {
                               faad => {
                                         NAME => "faad",
                                         DSTR => sub { "$dopts -o %i %o" },
                                       },
-                              
+
                             ffmpeg => {
                                         NAME => "ffmpeg",
                                         DSTR => sub { "$dopts -y -i %i %o" },
@@ -977,27 +987,27 @@
 
                 DEFAULT_ENCODER => "faac",
                 DEFAULT_DECODER => "faad",
-                
+
                 ENCODER => {
-                
+
                             faac => {
                                       NAME => "faac",
-                                      ESTR => sub { 
+                                      ESTR => sub {
                                                     $opts = "-q $aacqual" if $defopts == 1;
                                                     $opts = '' if $defopts == 0;
-                                                    "$eopts -w $opts %i -o %o" 
+                                                    "$eopts -w $opts %i -o %o"
                                                   },
                                       PROMPT => { AACQUAL => 1 },
                                     },
 
                           ffmpeg => {
                                       NAME => "ffmpeg",
-                                      ESTR => sub { 
+                                      ESTR => sub {
                                                     $opts = "-ab $bitrate.k -ar $freq -ac $channels" if $defopts == 1;
                                                     $opts = '' if $defopts == 0;
-                                                    "$eopts -y -i %i $opts %o" 
+                                                    "$eopts -y -i %i $opts %o"
                                                   },
-                                      
+
                                       PROMPT => {
                                                   BITRATE  => 1,
                                                   FREQ     => 1,
@@ -1005,7 +1015,7 @@
                                                 },
                                     },
                            },
-                             
+
                 DECODER => {
 
                             faad => {
@@ -1013,7 +1023,7 @@
                                       DSTR => sub { "$dopts -o %o %i" },
                                     },
 
-                          ffmpeg => { 
+                          ffmpeg => {
                                       NAME => "ffmpeg",
                                       DSTR => sub { "$dopts -y -i %i %o" },
                                     },
@@ -1035,27 +1045,27 @@
 
                 DEFAULT_ENCODER => "faac",
                 DEFAULT_DECODER => "faad",
-                
+
                 ENCODER => {
-                  
+
                             faac => {
                                       NAME => "faac",
-                                      ESTR => sub { 
+                                      ESTR => sub {
                                                     $opts = "-q $aacqual" if $defopts == 1;
                                                     $opts = '' if $defopts == 0;
-                                                    "$eopts -w $opts %i -o %o" 
+                                                    "$eopts -w $opts %i -o %o"
                                                   },
                                       PROMPT => { AACQUAL => 1 },
                                     },
 
                           ffmpeg => {
                                       NAME => "ffmpeg",
-                                      ESTR => sub { 
+                                      ESTR => sub {
                                                     $opts = "-ab $bitrate.k -ar $freq -ac $channels" if $defopts == 1;
                                                     $opts = '' if $defopts == 0;
-                                                    "$eopts -y -i %i $opts %o" 
+                                                    "$eopts -y -i %i $opts %o"
                                                   },
-                                      
+
                                       PROMPT => {
                                                   BITRATE  => 1,
                                                   FREQ     => 1,
@@ -1063,16 +1073,16 @@
                                                 },
                                     },
                            },
-                
+
                 DECODER => {
 
                              faad => {
                                        NAME => "faad",
                                        DSTR => sub { "$dopts -o %o %i" },
                                      },
-                                       
+
                            ffmpeg => {
-                                       NAME => "ffmpeg", 
+                                       NAME => "ffmpeg",
                                        DSTR => sub { "$dopts -y -i %i %o" },
                                      },
 
@@ -1080,7 +1090,7 @@
                                        NAME => "mplayer",
                                        DSTR => sub { "-vc dummy -vo null -nortc -ao pcm:file=%o %i" },
                                      },
-                                      
+
                            },
 
                 ESTR    => sub { "$eopts -w -q $aacqual %i -o %o" },
@@ -1093,12 +1103,12 @@
                },
 
        m4b  => {
-       
+
                 DEFAULT_ENCODER => "faac",
                 DEFAULT_DECODER => "faad",
-                
+
                 ENCODER => {
-                           
+
                             faac => {
                                       NAME => "faac",
                                       ESTR => sub {
@@ -1109,7 +1119,7 @@
 
                                       PROMPT => { AACQUAL => 1 },
                                     },
- 
+
                            },
 
                 DECODER => {
@@ -1125,32 +1135,32 @@
                            },
 
                 ESTR    => sub { "$eopts -w -q $aacqual %i -o %o" },
-                
+
                 TAGS    => {
                              READ   => 0,
                              WRITE  => 1,
                              MODULE => undef,
                            },
                },
-                             
+
        mpc  => {
 
                 DEFAULT_ENCODER => "mppenc",
                 DEFAULT_DECODER => "mppdec",
-                
+
                 ENCODER => {
 
                             mppenc => {
                                         NAME => "mppenc",
-                                        ESTR => sub { 
+                                        ESTR => sub {
                                                       $opts = "--$mpcqual" if $defopts == 1;
                                                       $opts = '' if $defopts == 0;
-                                                      "$eopts --overwrite $opts %i %o" 
+                                                      "$eopts --overwrite $opts %i %o"
                                                     },
                                         PROMPT => { MPCQUAL => 1 },
                                       },
                            },
-                
+
                 DECODER => {
 
                             mppdec => {
@@ -1179,20 +1189,20 @@
 
                 DEFAULT_ENCODER => "mppenc",
                 DEFAULT_DECODER => "mppdec",
-                
+
                 ENCODER => {
 
                             mppenc => {
                                         NAME => "mppenc",
-                                        ESTR => sub { 
+                                        ESTR => sub {
                                                       $opts = "--$mpcqual" if $defopts == 1;
                                                       $opts = '' if $defopts == 0;
-                                                      "$eopts --overwrite $opts %i %o" 
+                                                      "$eopts --overwrite $opts %i %o"
                                                     },
                                         PROMPT => { MPCQUAL => 1 },
                                       },
                            },
-                
+
                 DECODER => {
 
                             mppdec => {
@@ -1217,23 +1227,23 @@
 
                 DEFAULT_ENCODER => "ofr",
                 DEFAULT_DECODER => "ofr",
-                
+
                 ENCODER => {
                               ofr => {
                                         NAME => "ofr",
-                                        ESTR => sub { 
+                                        ESTR => sub {
                                                       $opts = "--mode $ofmode --optimize $ofopt" if $defopts == 1;
                                                       $opts = '' if $defopts == 0;
-                                                      "$eopts --overwrite $opts %i --output %o" 
+                                                      "$eopts --overwrite $opts %i --output %o"
                                                     },
-                                        
+
                                         PROMPT => {
                                                     OFMODE => 1,
                                                     OFOPT  => 1,
                                                   },
                                      },
                            },
-                           
+
                 DECODER => {
                               ofr => {
                                         NAME => "ofr",
@@ -1252,23 +1262,23 @@
 
                 DEFAULT_ENCODER => "ofs",
                 DEFAULT_DECODER => "ofs",
-                
+
                 ENCODER => {
                               ofs => {
                                         NAME => "ofs",
-                                        ESTR => sub { 
+                                        ESTR => sub {
                                                       $opts = "--mode $ofmode --optimize $ofopt" if $defopts == 1;
                                                       $opts = '' if $defopts == 0;
-                                                      "$eopts --overwrite $opts %i --output %o" 
+                                                      "$eopts --overwrite $opts %i --output %o"
                                                     },
-                                    
+
                                         PROMPT => {
                                                     OFMODE => 1,
                                                     OFOPT  => 1,
                                                   },
                                      },
                            },
-                
+
                 DECODER => {
                               ofs => {
                                         NAME => "ofs",
@@ -1287,20 +1297,20 @@
 
                 DEFAULT_ENCODER => "lpac",
                 DEFAULT_DECODER => "lpac",
-                
+
                 ENCODER => {
                               lpac => {
                                         NAME => "lpac",
-                                        ESTR => sub { 
+                                        ESTR => sub {
                                                       $opts = "-$pcomp" if $defopts == 1;
                                                       $opts = '' if $defopts == 0;
-                                                      "$eopts -v $opts %i %o" 
+                                                      "$eopts -v $opts %i %o"
                                                     },
-                                      
+
                                         PROMPT => { PCOMP => 1 },
                                       },
                            },
-                
+
                 DECODER => {
                               lpac => {
                                         NAME => "lpac",
@@ -1319,20 +1329,20 @@
 
                 DEFAULT_ENCODER => "lpac",
                 DEFAULT_DECODER => "lpac",
-                
+
                 ENCODER => {
                               lpac => {
                                         NAME => "lpac",
-                                        ESTR => sub { 
+                                        ESTR => sub {
                                                       $opts = "-$pcomp" if $defopts == 1;
                                                       $opts = '' if $defopts == 0;
-                                                      "$eopts -v $opts %i %o" 
+                                                      "$eopts -v $opts %i %o"
                                                     },
-                                      
+
                                         PROMPT => { PCOMP => 1 },
                                       },
                            },
-                
+
                 DECODER => {
                               lpac => {
                                         NAME => "lpac",
@@ -1351,14 +1361,14 @@
 
                 DEFAULT_ENCODER => "ttaenc",
                 DEFAULT_DECODER => "ttaenc",
-                
+
                 ENCODER => {
                               ttaenc => {
                                           NAME => "ttaenc",
                                           ESTR => sub { "$eopts -e %i -o %o" },
                                         },
                            },
-                
+
                 DECODER => {
                               ttaenc => {
                                           NAME => "ttaenc",
@@ -1375,7 +1385,7 @@
                                           DSTR => sub { "-vc dummy -vo null -nortc -ao pcm:file=%o %i" },
                                         },
                            },
-                                        
+
                 TAGS    => {
                              READ   => 0,
                              WRITE  => 0,
@@ -1387,14 +1397,14 @@
 
                 DEFAULT_ENCODER => "la",
                 DEFAULT_DECODER => "la",
-                
+
                 ENCODER => {
                               la => {
                                       NAME => "la",
                                       ESTR => sub { "-overwrite $eopts %i %o" },
                                     },
                            },
-                
+
                 DECODER => {
                               la => {
                                       NAME => "la",
@@ -1413,14 +1423,14 @@
 
                 DEFAULT_ENCODER => "wavpack",
                 DEFAULT_DECODER => "wvunpack",
-                
+
                 ENCODER => {
                               wavpack => {
                                            NAME => "wavpack",
                                            ESTR => sub { "$eopts -y %i -o %o" },
                                          },
                            },
-                
+
                 DECODER => {
                               wvunpack => {
                                             NAME => "wvunpack",
@@ -1449,14 +1459,14 @@
 
                 DEFAULT_ENCODER => "mv",
                 DEFAULT_DECODER => "cp",
-                
+
                 ENCODER => {
                               mv => {
                                       NAME => "mv",
                                       ESTR => sub { "%i %o" },
                                     },
                            },
-                           
+
                 DECODER => {
                               cp => {
                                       NAME => "cp",
@@ -1497,13 +1507,13 @@
 # support reading of WMA tags using the Audio::WMA module
 $run{wma}{TAGS}{READ}   = 1;
 $run{wma}{TAGS}{MODULE} = "Audio::WMA";
-   
+
 # supported video to audio types
-my @movie_codecs = ( 
-                     "rv",    "asf",   "avi",   "divx",  
-                     "mkv",   "mpg",   "mpeg",  "mov",   
-                     "ogm",   "qt",    "vcd",   "vob",   
-                     "wmv",   "flv",   "svcd",  "m4v",   
+my @movie_codecs = (
+                     "rv",    "asf",   "avi",   "divx",
+                     "mkv",   "mpg",   "mpeg",  "mov",
+                     "ogm",   "qt",    "vcd",   "vob",
+                     "wmv",   "flv",   "svcd",  "m4v",
                      "nsv",   "nuv",   "psp",   "smk",
                    );
 
@@ -1521,13 +1531,13 @@
 }
 
 # supported sndfile-convert formats
-my @sndfile = ( 
-                "pvf",  "caf",  "sf",  "paf",   "fap", "sd2", 
+my @sndfile = (
+                "pvf",  "caf",  "sf",  "paf",   "fap", "sd2",
                 "mat4", "mat5", "mat", "ircam", "w64", "nist",
               );
 
 foreach (@sndfile) {
-     
+
    $run{$_}{DEFAULT_ENCODER}                  = "sndfile-convert";
    $run{$_}{DEFAULT_DECODER}                  = "sndfile-convert";
    $run{$_}{ENCODER}{"sndfile-convert"}{NAME} = "sndfile-convert";
@@ -1538,7 +1548,7 @@
    $run{$_}{TAGS}{WRITE}                      = 0;
    $run{$_}{TAGS}{MODULE}                     = undef;
 }
-              
+
 # supported sox formats
 my @sox_codecs = ( "aiff", "aif", "au", "snd", "voc", "smp", "avr", "cdr" );
 
@@ -1562,7 +1572,7 @@
 # load codecs.conf file
 load_codecs() if -e $codecs_conf;
 
-my $out_name = $outfile; 
+my $out_name = $outfile;
 my $out_dir  = $outdir;
 
 my $total_converted = 0;
@@ -1573,14 +1583,14 @@
 sub proc_input {
 
    $to =~ tr/A-Z/a-z/;
-   
+
    perror("bad_input","") unless(exists($run{$to}));
-   
+
    # process all files first
    if (@file) {
 
        foreach my $i (sort(@file)) {
-       
+
                my ($file, $dir, $ext) = fileparse("$i", qr/\.[^.]*/);
 
                    $ext    =~ s/^\.//;
@@ -1590,26 +1600,26 @@
 
                my  $if  = $ext;
                    $if  =~ tr/A-Z/a-z/;
-                   
+
                    $only =~ tr/A-Z/a-z/ if $only;
-                   
+
               next if $only and $if ne $only;
-              
+
                    $encoder = $my_encoder if defined($my_encoder);
                    $decoder = $my_decoder if defined($my_decoder);
-                   
+
                    $encoder = $run{$to}{DEFAULT_ENCODER} if not $my_encoder;
                    $decoder = $run{$if}{DEFAULT_DECODER} if not $my_decoder;
-                                      
+
                if (check_input($file,$dir,$if) == 0) {
-                   
+
                    pnotice("unk_encoder","$file.$ext",1), next unless(exists($run{$to}{ENCODER}{$encoder}));
                    pnotice("unk_decoder","$file.$ext",1), next unless(exists($run{$if}{DECODER}{$decoder}));
-              
+
                    get_visual_opts($to,$outdir) if $gui and not defined($first_run) and `which kdialog 2>/dev/null`;
-                   
+
                    $first_run = 1;
-                   
+
                    perror("multi_out","")       if (defined($out_name) and $#file gt 0 or defined($out_name) and @dir);
                    perror("no_outdir","")       if (not -e $outdir and not -d $outdir);
 
@@ -1618,12 +1628,12 @@
                    if ($keep and $if eq $to) {
                        system("cp -v \"$dir/$file.$ext\" \"$outdir/$outfile.$ext\"") if $outdir ne $dir and not -e "$outdir/$outfile.$ext";
                        next if $outdir eq $dir;
-                   } 
-                   
+                   }
+
                    else {
                        convert("$dir/$file","$outdir/$outfile","$ext","$file","$outfile");
                    }
-                 
+
                }
        }
    }
@@ -1638,13 +1648,13 @@
     if (not $recursive) {
 
         undef(@file);
-        
+
         foreach my $d (@tmp_dir) {
 
                 opendir(DIR, "$d") or perror("opening_dir","$d: $!");
              my @dir_files = readdir(DIR);
                 closedir(DIR);
-                
+
                 perror("empty_dir","$d") if $#dir_files <= 1;
 
                 foreach my $f (@dir_files) {
@@ -1655,16 +1665,16 @@
                      }
                }
        }
-       
+
        proc_input();
        return 0;
 
-    } 
+    }
 
     elsif ($recursive) {
 
              if (not $preserve) {
-                 
+
                  undef(@file);
 
                  foreach my $r (@tmp_dir) {
@@ -1677,7 +1687,7 @@
                                     }, $r
                               );
                  }
-                 
+
                  proc_input();
                  return 0;
 
@@ -1689,7 +1699,7 @@
       }
 
     }
-   
+
     print "\n$lang{total_converted} $total_converted, $lang{failed} $total_failed\n";
 }
 
@@ -1721,7 +1731,7 @@
         if (-d $find_dir and not -d $odir) { mkdir("$odir", 0755); }
 
             opendir(IN, "$presdir") or perror("opening_dir","$presdir: $!");
-            
+
             undef(@file);
 
             foreach my $i (readdir(IN)) {
@@ -1733,7 +1743,7 @@
                  }
 
             } undef($presdir); undef($odir); closedir(IN);
-            
+
             proc_input();
 
         } elsif (-d $find_dir and $find_dir eq $topdir) {
@@ -1741,7 +1751,7 @@
                  $presdir = $topdir;
                  $odir    = $out_copy;
                  $out_dir = $odir;
-                 
+
                  undef(@file);
 
                  opendir(IN, "$presdir") or perror("opening_dir","$presdir: $!");
@@ -1753,7 +1763,7 @@
                       }
 
                  } closedir(IN);
-                 
+
                  proc_input();
           }
 }
@@ -1777,21 +1787,21 @@
        # catch ^C.  will have to be pressed repeatedly to exit the process.
        $SIG{INT} = sub {
                          unlink("$outf.$$.wav");
-                         kill 9, $$; 
+                         kill 9, $$;
                        };
-       
-       # check to see if encoder/decoder exists.  if not, see if we have one 
+
+       # check to see if encoder/decoder exists.  if not, see if we have one
        # that supports the desired input/output formats.
        check_encoder();
        check_decoder($infmt);
-                         
+
     my $decode_string = $run{$if}{DECODER}{$decoder}{DSTR}->();
        $decode_string =~ s/%i/\"$inf.$infmt\"/;
        $decode_string =~ s/%o/\"$outf.$$.wav\"/;
        $decode_string =~ s/(\$|\\)/\\$1/g;
 
     my $nl = 0; $nl = 2 if $verbose;
-          
+
        pnotice("converting"," [$iname.$infmt] -> [$to] ", $nl);
        pnotice("debug","$run{$if}{DECODER}{$decoder}{NAME} $decode_string", 2) if $dryrun;
 
@@ -1805,21 +1815,21 @@
            $total_failed++;
            return 0;
        }
-       
+
        # normalize wav file before encoding to output format
        perror("no_app","normalize") if $normalize and not `which normalize 2>/dev/null`;
        system("normalize $nopts \"$outf.$$.wav\"") if $normalize;
 
-       # copy meta-data from input file to %tag_name 
+       # copy meta-data from input file to %tag_name
        read_tags("$inf.$infmt", "$if") if $run{$if}{TAGS}{READ} == 1 and not $dryrun;
 
     my $encode_string = $run{$to}{ENCODER}{$encoder}{ESTR}->();
-       $encode_string =~ s/%i/\"$outf.$$.wav\"/; 
+       $encode_string =~ s/%i/\"$outf.$$.wav\"/;
        $encode_string =~ s/%o/\"$outf.$to\"/;
        $encode_string =~ s/(\$|\\)/\\$1/g;
 
     my $tag_opts = format_tags($to); # tag options for mp4/m4a/m4b/mpc/mpp/bonk/spx/wv
-    
+
        # read tag options from command line if present
        get_user_tags();
 
@@ -1839,13 +1849,13 @@
 
        # write meta-data to output file from %tag_name hash
        write_tags("$outf.$to", "$to") if $run{$to}{TAGS}{WRITE} == 1 and $tag_opts eq '' and not $dryrun;
- 
+
        unlink("$outf.$$.wav") if not $dryrun;
        unlink("$inf.$infmt")  if $delete and not $dryrun;
 
        pnotice("removed_tmp","$outf.$$.wav","2") if $verbose;
        pnotice("removed_src","$inf.$infmt", "2") if $delete and $verbose;
-       
+
        if (-e "$outf.$to") {
            pnotice("conv_comp","",1);
            $total_converted++;
@@ -1858,9 +1868,9 @@
 
     if (not `which $encoder 2>/dev/null`) {
         my $first_encoder = $encoder;
-        foreach (keys %{$run{$to}{ENCODER}}) { $encoder = $_ and return if `which $_ 2>/dev/null`; } 
+        foreach (keys %{$run{$to}{ENCODER}}) { $encoder = $_ and return if `which $_ 2>/dev/null`; }
         perror("no_encode_app","$to") if $encoder eq $first_encoder;
-    } 
+    }
 
     else { return 0; }
 }
@@ -1870,13 +1880,13 @@
 sub check_decoder {
 
     my $from = shift;
-    
+
     if (not `which $decoder 2>/dev/null`) {
         my $first_decoder = $decoder;
         foreach (keys %{$run{$from}{DECODER}}) { $decoder = $_ and return if `which $_ 2>/dev/null`; }
         perror("no_decode_app","$from") if $decoder eq $first_decoder;
     }
-   
+
     else { return 0; }
 }
 
@@ -1912,7 +1922,7 @@
        chomp($outdir);
        $out_dir = $outdir;
        exit(1) if not $outdir;
-       
+
        if ($config{KDE_OPTS} == 1) {
            foreach (keys %{$run{$to_fmt}{ENCODER}{$encoder}{PROMPT}}) {
                    prompt_user($_);
@@ -1926,7 +1936,7 @@
 
       switch ($opt) {
 
-            case 'BITRATE'   { 
+            case 'BITRATE'   {
                                 $bitrate  = `$kdialog --default $bitrate  --combobox \"$lang{kde_bitrate}\" 56 112 128 160 192 256 320`;
                                 chomp($bitrate);
                                 exit(1) if not $bitrate;
@@ -1938,7 +1948,7 @@
                                 exit(1) if not $freq;
                              }
 
-            case 'CHANNELS'  { 
+            case 'CHANNELS'  {
                                 $channels = `$kdialog --default $channels --combobox \"$lang{kde_chans}\" 1 2`;
                                 chomp($channels);
                                 exit(1) if not $channels;
@@ -1967,7 +1977,7 @@
                                chomp($oggqual);
                                exit(1) if not $oggqual;
                              }
-                             
+
             case 'SPXQUAL'   {
                                $spxqual  = `$kdialog --default $spxqual   --combobox \"$lang{kde_spxqual}\" 0 1 2 3 4 5 6 7 8 9 10`;
                                chomp($spxqual);
@@ -1987,13 +1997,13 @@
                              }
 
             case 'OFMODE'    {
-                               $ofmode   = `$kdialog --default $ofmode    --combobox \"$lang{kde_ofmode}\" fast normal high extra best highnew extranew bestnew`; 
+                               $ofmode   = `$kdialog --default $ofmode    --combobox \"$lang{kde_ofmode}\" fast normal high extra best highnew extranew bestnew`;
                                chomp($ofmode);
                                exit(1) if not $ofmode;
                              }
 
             case 'OFOPT'     {
-                               $ofopt    = `$kdialog --default $ofopt     --combobox \"$lang{kde_ofopt}\" none fast normal high best`; 
+                               $ofopt    = `$kdialog --default $ofopt     --combobox \"$lang{kde_ofopt}\" none fast normal high best`;
                                chomp($ofopt);
                                exit(1) if not $ofopt;
                              }
@@ -2043,33 +2053,33 @@
     $tag_name{comment} = $comment if $comment;
     $tag_name{year}    = $year    if $year;
     $tag_name{genre}   = $genre   if $genre;
-    
+
 }
 
 # the formats MP4/M4A/M4B/MPC/MPP/WV/BONK have no tag writing module.
-# tags therefore have to be written during encode via various 
+# tags therefore have to be written during encode via various
 # command line args.
 sub format_tags {
 
     my $type = shift;
 
     switch ($type) {
-        
-        case /mp4|m4a|m4b/ { 
-        
+
+        case /mp4|m4a|m4b/ {
+
             if ($run{$to}{ENCODER}{$encoder}{NAME} eq "ffmpeg") {
-     
+
                 $tag_name{track} = 0 if not $tag_name{track};
                 $tag_name{year}  = 0 if not $tag_name{year};
                 return "-title \"$tag_name{title}\" -track \"$tag_name{track}\" -author \"$tag_name{artist}\" -album \"$tag_name{album}\" -comment \"$tag_name{comment}\" -year \"$tag_name{year}\" -genre \"$tag_name{genre}\"";
 
             } else {
-                
+
               return "--title \"$tag_name{title}\" --track \"$tag_name{track}\" --artist \"$tag_name{artist}\" --album \"$tag_name{album}\" --comment \"$tag_name{comment}\" --year \"$tag_name{year}\" --genre \"$tag_name{genre}\"";
 
               }
-        } 
-                
+        }
+
         case /mpc|mpp/ { return "--tag \"Title=$tag_name{title}\" --tag \"Track=$tag_name{track}\" --tag \"Artist=$tag_name{artist}\" --tag \"Album=$tag_name{album}\" --tag \"Comment=$tag_name{comment}\" --tag \"Year=$tag_name{year}\" --tag \"Genre=$tag_name{genre}\""; }
         case 'wv'      { return "-w \"Title=$tag_name{title}\" -w \"Track=$tag_name{track}\" -w \"Artist=$tag_name{artist}\" -w \"Album=$tag_name{album}\" -w \"Comment=$tag_name{comment}\" -w \"Year=$tag_name{year}\" -w \"Genre=$tag_name{genre}\""; }
         case 'bonk'    { return "-t \"$tag_name{title}\" -a \"$tag_name{artist}\" -c \"$tag_name{comment}\""; }
@@ -2079,7 +2089,7 @@
 }
 
 # meta-data is read based on input file extension.
-# all tags which are present will be copied to 
+# all tags which are present will be copied to
 # the %tag_name hash.
 sub read_tags {
 
@@ -2099,7 +2109,7 @@
                                $tag_name{comment} = "$tags[4]" if $tags[4];
                                $tag_name{year}    = "$tags[5]" if $tags[5];
                                $tag_name{genre}   = "$tags[6]" if $tags[6];
-                               
+
                             return 0;
                        }
 
@@ -2126,32 +2136,32 @@
                                  $tag_name{year}    = ${$yea} if defined(${$yea});
                                  $tag_name{genre}   = ${$gen} if defined(${$gen});
                             }
-                            
+
                             return 0;
                        }
-                       
+
         case 'spx'     {
                             my $pid  = open(SPX, "speexdec \"$in_file\" tmp1-$$.wav 2>&1 |") or perror("opening_file","$in_file");
                             my @data = <SPX>;
                                @data = grep(!/^$|^#/, @data);
-                            
+
                             close(SPX);
-                               
+
                             foreach (@data) {
 
-                                next unless /=/; 
-                               
+                                next unless /=/;
+
                                 chomp;
-                               
+
                                 my ($k, $v) = split /=/;
                                     $k =~ tr/A-Z/a-z/;
                                     $tag_name{$k} = $v if exists($tag_name{$k});
                                     $tag_name{artist} = $v if $k =~ /^author/i;
-                               
+
                              }
-                            
+
                              unlink("tmp-$$.wav");
-                             
+
                              return 0;
                        }
 
@@ -2194,7 +2204,7 @@
                                $tag_name{comment} = $mpc_tag->{COMMENT} if $mpc_tag->{COMMENT};
                                $tag_name{year}    = $mpc_tag->{YEAR}    if $mpc_tag->{YEAR};
                                $tag_name{genre}   = $mpc_tag->{GENRE}   if $mpc_tag->{GENRE};
-                               
+
                             return 0;
                        }
 
@@ -2208,7 +2218,7 @@
                                $tag_name{comment} = $wv_tag->{COMMENT} if $wv_tag->{COMMENT};
                                $tag_name{year}    = $wv_tag->{YEAR}    if $wv_tag->{YEAR};
                                $tag_name{genre}   = $wv_tag->{GENRE}   if $wv_tag->{GENRE};
-                            
+
                             return 0;
                        }
 
@@ -2222,7 +2232,7 @@
                                $tag_name{comment} = $wma_tag->{DESCRIPTION} if $wma_tag->{DESCRIPTION};
                                $tag_name{year}    = $wma_tag->{YEAR}        if $wma_tag->{YEAR};
                                $tag_name{genre}   = $wma_tag->{GENRE}       if $wma_tag->{GENRE};
-                            
+
                             return 0;
                        }
 
@@ -2240,7 +2250,7 @@
 
            case 'mp3'  {
                             $tag_m = MP3::Tag->new("$out_file");
-                            
+
                             # ID3v2 Tags
                             unless(exists($tag_m->{ID3v2})) { $tag_m->new_tag("ID3v2"); }
 
@@ -2253,10 +2263,10 @@
                             $tag_m->{ID3v2}->add_frame("TCON", "$tag_name{genre}")    if defined($tag_name{genre});
 
                             $tag_m->{ID3v2}->write_tag;
-                            
+
                             # ID3v1 Tags
                             unless(exists($tag_m->{ID3v1})) { $tag_m->new_tag("ID3v1"); }
-                            
+
                             $tag_m->{ID3v1}->title("$tag_name{title}")      if defined($tag_name{title});
                             $tag_m->{ID3v1}->track("$tag_name{track}")      if defined($tag_name{track});
                             $tag_m->{ID3v1}->artist("$tag_name{artist}")    if defined($tag_name{artist});
@@ -2264,19 +2274,19 @@
                             $tag_m->{ID3v1}->comment("$tag_name{comment}")  if defined($tag_name{comment});
                             $tag_m->{ID3v1}->year("$tag_name{year}")        if defined($tag_name{year});
                             $tag_m->{ID3v1}->genre("$tag_name{genre}")      if defined($tag_name{genre});
-                            
+
                             $tag_m->{ID3v1}->write_tag;
-                            
+
                             return 0;
 
                         }
 
            case 'ogg'   {
                             $tag_m = Ogg::Vorbis::Header->new("$out_file");
-                            
+
                             # this prevents duplicate tags =)
                             $tag_m->clear_comments();
-                            
+
                             # write new tags
                             $tag_m->add_comments("TITLE",       "$tag_name{title}")    if $tag_name{title};
                             $tag_m->add_comments("TRACKNUMBER", "$tag_name{track}")    if $tag_name{track};
@@ -2287,7 +2297,7 @@
                             $tag_m->add_comments("GENRE",       "$tag_name{genre}")    if $tag_name{genre};
 
                             $tag_m->write_vorbis;
-                            
+
                             return 0;
 
                         }
@@ -2305,7 +2315,7 @@
                             $tag_i->{GENRE}       = "$tag_name{genre}"   if $tag_name{genre};
 
                             $tag_m->write();
-                            
+
                           return 0;
                        }
 
@@ -2315,7 +2325,7 @@
 }
 
 # this is necessary when converting multiple files.
-# it prevents old tags from the previous file being 
+# it prevents old tags from the previous file being
 # copied over to the next file in line.
 sub clear_tag_hash {
     foreach (keys(%tag_name)) {
@@ -2331,9 +2341,9 @@
     foreach my $i (@file) {
 
      my ($file, $dir, $ext) = fileparse("$i", qr/\.[^.]*/);
-     
+
          $ext  =~ s/^\.//;
-     
+
      my  $from = $ext;
          $from =~ tr/A-Z/a-z/;
 
@@ -2342,7 +2352,7 @@
          clear_tag_hash();
          read_tags("$dir/$file.$ext","$from");
 
-         print "\n"; 
+         print "\n";
 
          pnotice("tag_info","$file.$ext",2);
 
@@ -2366,7 +2376,7 @@
     perror("no_input","") if $#file < 0;
 
     print "\n$name - $version\n\n";
-    
+
     foreach (@file) {
 
          my ($file, $dir, $ext) = fileparse("$_", qr/\.[^.]*/);
@@ -2389,7 +2399,7 @@
              $tag_name{genre}   = $genre   if $genre;
 
              write_tags("$dir/$file.$ext","$from");
-             print "$lang{write_tag_i} $file.$ext\n";         
+             print "$lang{write_tag_i} $file.$ext\n";
 
          } else { perror("no_write_tag","$file.$ext"); }
     }
@@ -2525,9 +2535,9 @@
         $to =~ tr/A-Z/a-z/;
         get_visual_opts($to,$out_dir) if $gui and not defined($first_run) and `which kdialog 2>/dev/null`;
         $first_run = 1;
-        
-        pnotice("ripping_track","$tno",2);    
- 
+
+        pnotice("ripping_track","$tno",2);
+
         $on =~ s/\//_/g;
 
         print  "cdparanoia -d $device $tno $on.wav\n\n"  if $dryrun;
@@ -2555,7 +2565,7 @@
 
  my $total_tracks = get_total_tracks();
 
-    if ($rip eq "all") { 
+    if ($rip eq "all") {
 
         if ($config{USE_CDDB} ne 1) {
 
@@ -2570,7 +2580,7 @@
               foreach my $i (@cdtrack) {
 
                       my $out_string = $nscheme;
-                      
+
                          chomp($cdtrack[$n-1]);
 
                          $out_string =~ s/%ar/$cd{artist}/;
@@ -2738,13 +2748,13 @@
         perror("opening_file","$file: $@ $!") if $@ or !defined($return) or not !$return;
     }
 }
-                                                 
+
 sub import_module {
     my ($format_ref) = @_;
     my $format_name=$format_ref->{NAME};
        $run{$format_name} = $format_ref;
 }
-               
+
 # load all user modules
 sub load_user_modules {
    my @imported_mods;
@@ -2762,13 +2772,13 @@
   foreach my $in (@ARGV) {
        if    (-f $in) { push(@file,$in); }
        elsif (-d $in) { push(@dir, $in); }
-       else { perror("no_infile","$in"); } 
+       else { perror("no_infile","$in"); }
   }
 }
 
 load_user_modules() if $config{IMPORTM} == 1;
 
-if ($to and @ARGV and not $rip) { proc_input(); } 
+if ($to and @ARGV and not $rip) { proc_input(); }
 
 elsif ($to and $rip) { proc_cd(); }
 elsif ($my_encoder and not $to) { show_encoders(); }
@@ -2819,7 +2829,7 @@
 
 B<-p, --preserve>
 
-when recursively converting a directory, preserve the input folders directory 
+when recursively converting a directory, preserve the input folders directory
 structure in the specified output directory.  such as:
 
 pacpl --to ogg -r -p /home/mp3s --outdir /home/oggs
@@ -2866,7 +2876,7 @@
 
 B<-k, --keep>
 
-do not re-encode files with extensions matching the destination extension.  
+do not re-encode files with extensions matching the destination extension.
 instead...copy them to the output folder, or skip to the next file.
 
 =over 4
@@ -2885,7 +2895,7 @@
 
 B<--defopts> I<1/0>
 
-to turn off default encoder options use --defopts 0. this option gives you 
+to turn off default encoder options use --defopts 0. this option gives you
 more control when using the --eopts command.
 
 defopts is set to 1 by default.  you can also toggle this option in
@@ -2960,7 +2970,7 @@
 
 B<--bitrate> I<num>
 
-set the bitrate to I<num>.  default is 128.  this option does not apply 
+set the bitrate to I<num>.  default is 128.  this option does not apply
 to all formats.
 
 B<--freq> I<num>
@@ -3104,7 +3114,7 @@
 
 =item best
 
-=back             
+=back
 
 B<--bratio> I<num>
 
@@ -3180,7 +3190,7 @@
 
 =item pacpl --rip all --to flac
 
-=item pacpl --rip 1,3,9,15 --to flac 
+=item pacpl --rip 1,3,9,15 --to flac
 
 =back
 
@@ -3218,7 +3228,7 @@
 
 set device to I<str>.  default is /dev/cdrom
 
-B<--cdinfo> 
+B<--cdinfo>
 
 show cddb information for current disc.
 
