Index: pacpl-4.0.5/codecs.conf
===================================================================
--- pacpl-4.0.5.orig/codecs.conf	2011-01-05 18:54:24.000000000 +0100
+++ pacpl-4.0.5/codecs.conf	2011-01-05 18:55:06.000000000 +0100
@@ -7,6 +7,10 @@
 #
 AAC = faac,faad
 
+# encoders: jmac, mac
+# decoders: jmac, mac
+APE = jmac,jmac
+
 # encoders: flac, ffmpeg
 # decoders: flac, ffmpeg, mplayer
 #
Index: pacpl-4.0.5/pacpl
===================================================================
--- pacpl-4.0.5.orig/pacpl	2011-01-05 18:54:32.000000000 +0100
+++ pacpl-4.0.5/pacpl	2011-01-05 18:56:12.000000000 +0100
@@ -823,6 +823,13 @@
                                                     },
                                         PROMPT => { ACOMP => 1 },
                                      },
+                              jmac => {
+                                        NAME => "java",
+                                        ESTR => sub {
+                                                      "-jar /usr/share/java/jmac.jar c2 %i %o"
+                                                    },
+                                     },
+
                             },
                 
                 DECODER => {
@@ -830,7 +837,11 @@
                                         NAME => "mac",
                                         DSTR => sub { "%i %o -d $dopts" },
                                      },
-                           },
+                              jmac => {
+                                        NAME => "java",
+                                        DSTR => sub { "-jar /usr/share/java/jmac.jar d %i %o" },
+                                     },
+			   },
 
                 TAGS    => {
                              READ   => 0,
