Index: git/codecs.conf
===================================================================
--- git.orig/codecs.conf	2014-10-12 22:43:45.445683836 +0200
+++ git/codecs.conf	2014-10-12 22:43:45.441683764 +0200
@@ -71,6 +71,8 @@
 # E: avconv, ffmpeg, sox, toolame, twolame
 # D: avconv, ffmpeg, mplayer 
 
+MP2 = ffmpeg,ffmpeg
+
 # OGG #
 # E: avconv, ffmpeg, oggenc, sox
 # D: avconv, ffmpeg, mplayer, oggdec, sox
Index: git/pacpl
===================================================================
--- git.orig/pacpl	2014-10-12 22:43:45.445683836 +0200
+++ git/pacpl	2014-10-12 22:43:45.445683836 +0200
@@ -1116,9 +1116,9 @@
                            },
   
                    TAGS => {
-                             READ   => 0,
-                             WRITE  => 0,
-                             MODULE => undef,
+                             READ   => 1,
+                             WRITE  => 1,
+                             MODULE => "MP3::Tag",
                            },
                },                                        
 			   
