Index: pacpl-4.0.5.ppa/codecs.conf
===================================================================
--- pacpl-4.0.5.ppa.orig/codecs.conf	2011-09-06 23:07:54.000000000 +0200
+++ pacpl-4.0.5.ppa/codecs.conf	2011-09-06 23:07:54.000000000 +0200
@@ -28,6 +28,11 @@
 M4A = faac,faad
 MP4 = faac,faad
 
+# encoders: twolame
+# decoders: twolame
+#
+MP2 = twolame,twolame
+
 # encoders: lame, bladeenc, gogo, toolame, ffmpeg, sox
 # decoders: lame, ffmpeg, mplayer, sox
 #
Index: pacpl-4.0.5.ppa/pacpl
===================================================================
--- pacpl-4.0.5.ppa.orig/pacpl	2011-09-06 23:07:54.000000000 +0200
+++ pacpl-4.0.5.ppa/pacpl	2011-09-06 23:07:54.000000000 +0200
@@ -485,6 +485,58 @@
                            },
                },
                
+        mp2 => {
+
+                DEFAULT_ENCODER => "twolame",
+                DEFAULT_DECODER => "twolame",
+
+                ENCODER => {
+                           twolame => {
+                                        NAME => "twolame",
+                                        ESTR => sub {
+                                                      $opts = "-b $bitrate" if $defopts == 1;
+                                                      $opts = '' if $defopts == 0;
+                                                      "$eopts $opts %i %o"
+                                                    },
+
+                                        PROMPT => { BITRATE => 1 },
+                                      },
+
+                            ffmpeg => {
+                                        NAME => "ffmpeg",
+                                        ESTR => sub {
+                                                      $opts = "-ab $bitrate.k -ar $freq -ac $channels" if $defopts == 1;
+                                                      $opts = '' if $defopts == 0;
+                                                      "$eopts -y -i %i $opts %o"
+                                                    },
+
+                                        PROMPT => {
+                                                    BITRATE  => 1,
+                                                    FREQ     => 1,
+                                                    CHANNELS => 1,
+                                                  },
+                                      },
+                           },
+
+                DECODER => {
+                            ffmpeg => {
+                                        NAME => "ffmpeg",
+                                        DSTR => sub { "$dopts -y -i %i %o" },
+                                      },
+
+                           mplayer => {
+                                        NAME => "mplayer",
+                                        DSTR => sub { "-vc dummy -vo null -nortc -ao pcm:file=%o %i" },
+                                      },
+                           },
+
+                TAGS    => {
+                             READ   => 1,
+                             WRITE  => 1,
+                             MODULE => "MP3::Tag",
+                           },
+               },
+               
         mp3 => {
 
                 DEFAULT_ENCODER => "lame",
