Description: Use Ogg::Vorbis::Header::PurePerl
Author: Tim Retout <diocles@debian.org>
Bug-Debian: http://bugs.debian.org/655405

Index: git/pacpl
===================================================================
--- git.orig/pacpl	2012-09-30 00:26:07.000000000 +0200
+++ git/pacpl	2012-09-30 00:26:07.000000000 +0200
@@ -26,7 +26,7 @@
 use File::Spec::Functions qw(rel2abs);
 
 # Tagging modules
-use Ogg::Vorbis::Header;
+use Ogg::Vorbis::Header::PurePerl;
 use MP3::Tag;
 use Audio::FLAC::Header;
 use MP4::Info;
@@ -734,7 +734,7 @@
                 TAGS    => {
                              READ   => 1,
                              WRITE  => 1,
-                             MODULE => "Ogg::Vorbis::Header",
+                             MODULE => "Ogg::Vorbis::Header::PurePerl",
                            },
                },
        
@@ -2358,7 +2358,7 @@
                         }
 
            when ('ogg')   {
-                            $tag_m = Ogg::Vorbis::Header->new("$out_file");
+                            $tag_m = Ogg::Vorbis::Header::PurePerl->new("$out_file");
                             
                             # this prevents duplicate tags =)
                             $tag_m->clear_comments();
Index: git/configure
===================================================================
--- git.orig/configure	2012-09-30 00:26:07.000000000 +0200
+++ git/configure	2012-09-30 00:26:07.000000000 +0200
@@ -3039,7 +3039,7 @@
 			File::Find
 			File::Spec::Functions
 			CDDB_get
-			Ogg::Vorbis::Header
+			Ogg::Vorbis::Header::PurePerl
 			Audio::Musepack
 			Audio::Scan
 			Audio::WMA
