Index: git/pacpl
===================================================================
--- git.orig/pacpl	2012-09-30 00:26:07.000000000 +0200
+++ git/pacpl	2012-09-30 00:26:18.000000000 +0200
@@ -32,7 +32,7 @@
 use MP4::Info;
 use Audio::WMA;
 use Audio::Musepack;
-use Audio::APETags;
+use Audio::Scan;
 
 # CDDB module
 use CDDB_get qw(get_cddb);
@@ -1504,7 +1504,7 @@
                 TAGS    => {
                              READ   => 1,
                              WRITE  => 1,
-                             MODULE => "Audio::APETags",
+                             MODULE => "Audio::Scan",
                            },
                },
 
@@ -2285,7 +2285,7 @@
                        }
 
         when ('wv')      {
-                            my $wv_tag = $tag_module->getTags($in_file)->{'tags'};
+                            my $wv_tag = $tag_module->scan_tags($in_file)->{'tags'};
 
                                $tag_name{title}   = $wv_tag->{TITLE}   if $wv_tag->{TITLE};
                                $tag_name{track}   = $wv_tag->{TRACK}   if $wv_tag->{TRACK};
Index: git/configure
===================================================================
--- git.orig/configure	2012-09-30 00:26:07.000000000 +0200
+++ git/configure	2012-09-30 00:26:18.000000000 +0200
@@ -3041,7 +3041,7 @@
 			CDDB_get
 			Ogg::Vorbis::Header
 			Audio::Musepack
-			Audio::APETags
+			Audio::Scan
 			Audio::WMA
 			MP3::Tag
 			MP4::Info
Index: git/configure.ac
===================================================================
--- git.orig/configure.ac	2012-09-30 00:26:07.000000000 +0200
+++ git/configure.ac	2012-09-30 00:26:07.000000000 +0200
@@ -70,7 +70,7 @@
 			CDDB_get
 			Ogg::Vorbis::Header 
 			Audio::Musepack 
-			Audio::APETags 
+			Audio::Scan
 			Audio::WMA 
 			MP3::Tag
 			MP4::Info 
Index: git/extra/mod-install.sh
===================================================================
--- git.orig/extra/mod-install.sh	2012-09-30 00:25:45.000000000 +0200
+++ git/extra/mod-install.sh	2012-09-30 00:26:07.000000000 +0200
@@ -21,5 +21,5 @@
 $cpan 'install Ogg::Vorbis::Header'
 $cpan 'install IO::String'
 $cpan 'install MP4::Info'
-$cpan 'install Audio::APETags'
-$cpan 'install CDDB_get'
\ No newline at end of file
+$cpan 'install Audio::Scan'
+$cpan 'install CDDB_get'
